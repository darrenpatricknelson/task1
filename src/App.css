.app {
  text-align: center;
  width: 100%;
  height: 100vh;
  position: relative;
  background-color: rgba(0, 0, 0, 0.4);
  color: #fff;
  padding: 100px 0;
}

.app::before {
  content: '';
  background: url('./assets/backdrop.jpg') no-repeat center center/cover;
  position: absolute;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  z-index: -1;
}

/* input */
.input {
  width: 300px;
  height: 50px;
  margin: 15px 0 50px;
  text-align: center;
  border-radius: 1rem;
  border: 1px solid #fff;
  color: #fff;
  font-size: 1.5rem;
  background-color: transparent;
}

::placeholder {
  color: rgba(255, 255, 255, 0.5);
}
