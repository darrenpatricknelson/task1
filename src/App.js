import React, { useState } from 'react';
import 'isomorphic-fetch';
import './App.css';
import WeatherResults from './components/WeatherResults';
import Error from './components/Error';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      error: null,
      isLoaded: false,
      data: []
    };
    this.handleLocationSearch = this.handleLocationSearch.bind(this);
  }

  handleLocationSearch(event) {
    // this if statement is to check if the key press is the Enter key
    // Otherwise the block of code will run after each and every key press
    if (event.key === 'Enter') {
      // storing the users input (which will be the location) inside this variable
      // also replacing white spaces with '+' and convert all case to lower case
      //  the reason I am doing this is because the location variable will be put into the location part of the url on line 26
      const location = event.target.value.replaceAll(' ', '+').toLowerCase();

      // Now that I have a location that can be search along with the url
      // we can use this url in the fetch function
      const url = `https://api.openweathermap.org/data/2.5/weather?q=${location}&appid=${process.env.REACT_APP_MY_API_KEY}&units=metric`;

      fetch(url)
        .then((response) => {
          // I'm checking the status of the data package that we fetched
          // if it is no ok, It throws and error
          if (!response.ok) {
            throw new Error(respone.statusText);
          }
          return response.json();
        })
        .then((data) => {
          this.setState({
            isLoaded: true,
            data: data,
            error: null
          });
        })
        // that error is caught here
        // I know we were supposed to handle the error in the promise but I could not figure it out
        // All of my time spent researching error handling led me to the throw/catch method
        // So I implemented that and it works
        .catch((error) => {
          this.setState({
            isLoaded: true,
            error
          });
        });
    }
  }

  render() {
    return (
      <div className="app">
        {/* user input */}
        <div className="userInput">
          <input
            onKeyPress={this.handleLocationSearch}
            className="input"
            placeholder="Enter your city"
          />
        </div>

        {/* if error is true - display an error message */}
        {this.state.error && <Error messsage={this.state.errorMessage} />}
        {/* if the error is false - return a componenet display the weather data */}
        {this.state.isLoaded && !this.state.error && (
          <WeatherResults data={this.state.data} />
        )}
      </div>
    );
  }
}

export default App;
