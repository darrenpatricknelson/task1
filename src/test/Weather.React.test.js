import App from '../App';
import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(<App />).toJSON();

  expect(tree).toMatchSnapshot();
});

/* 
Hi Moderator 

So for this component test, I wanted to test out my 
"<WeatherResults data={this.state.data} />" component used in my app to display the weather results. 

However, since this component has props, I was not able to figure out how to test a component with props.

The failed test on the right ===> 
was the test with the WeatherResults component with its props.

How do I test a component with props? Is there a way to import the prop an store it as a variable?

I'm quite unsure about this so I to the safe option of testing my App component which is the successful test you see you at the bottom ===>

However, I am really curious to know how to test a component with props becaues it was not spoken about in the doc

Thank you in advance :)



*/
