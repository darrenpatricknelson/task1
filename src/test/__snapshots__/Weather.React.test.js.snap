// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="app"
>
  <div
    className="userInput"
  >
    <input
      className="input"
      onKeyPress={[Function]}
      placeholder="Enter your city"
    />
  </div>
</div>
`;
